python/tvm/meta_schedule/profiler.py

```diff
@@ -0,0 +1,72 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""A context manager that profiles tuning time cost for different parts."""
+from __future__ import annotations
+
+import logging
+from typing import Dict
+
+from tvm._ffi import register_object
+from tvm.runtime import Object
+
+from . import _ffi_api
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+class TimeItContext:
+    """The context to profile given scope."""
+
+    profiler: Profiler
```

Comment:
just curious, will it pass mypy/pylint type checking? if not, let's use string

```suggestion
    profiler: "Profiler"
```

Reply:
I have `from __future__ import annotations` which supports such type annotation. Is that something we can use?

Reply:
if the CI doesn't complain that will be fine :-) I thought we will need python 3.8 though

Reply:
Thanks for the advice, I got it tested locally on python3.8.8 and it seems to be working fine.